tests/conftest.py

```diff
@@ -1,213 +1,23 @@
-                'settings': foo_settings,
-                'dir': foo_dir_name},
-        'bar': {'mapping': bar_mapping,
-                'settings': bar_settings,
-                'dir': bar_dir_name},
-    }
-    root = tmp_path
-    return mock_mappings(root, mapps)
-
-
-def mock_mappings(tmp_path, mappings):
-    root = tmp_path
-    expected = {}
-    for mapping_name, data in mappings.items():
-        mapp_dir = root/data['dir']
-        mapp_dir.mkdir()
-        write_dict_to_yaml(mapp_dir, data['dir']+'.mapping.yaml',
-                           data.get('mapping', {}))
-
-        if data.get('settings'):
-            write_dict_to_yaml(mapp_dir, 'settings.yaml',
-                               data.get('settings', {}))
-
-        expected_mapping = copy.deepcopy(data.get('mapping', {}))
-        # descriptions has to be present in payload and descriptions need to
-        # have '_meta' key in them, otherwise they won't be picked up
-        if 'descriptions' in data and data['descriptions'].get('_meta'):
-            desc = data['descriptions']
-            write_dict_to_yaml(mapp_dir, 'descriptions.yaml', desc)
-            expected_mapping.update({'_meta': desc.get('_meta')})
-
-        expected[data['dir']] = {
-            'mapping': expected_mapping,
-            'settings': data.get('settings', {}),
-        }
-
-    return root, expected
-
-
-def get_generic_mapping():
-    foo_mapping = {
-        'properties': {
-            'foo': {'type': 'keyword'},
-            'bar': {'type': 'long'},
-        }
-    }
-    foo_dir_name = 'foo_descriptions'
-    return foo_mapping, foo_dir_name
-
-
-@pytest.fixture
-def descriptions_mappings(tmp_path):
-    foo_mapping, foo_dir_name = get_generic_mapping()
-    descriptions = {'_meta': 'expected_result'}
-    mapps = {
-        'foo': {
-            'mapping': foo_mapping,
-            'dir': foo_dir_name,
-            'descriptions': descriptions,
-        }
-    }
-    root = tmp_path
-    return mock_mappings(root, mapps)
 
 
 @pytest.fixture
-def no_descriptions_mappings(tmp_path):
-    foo_mapping, foo_dir_name = get_generic_mapping()
-    mapps = {
-        'foo': {
-            'mapping': foo_mapping,
-            'dir': foo_dir_name,
-        }
-    }
-    root = tmp_path
-    return mock_mappings(root, mapps)
-
+def es_models(tmp_path: pathlib.Path, monkeypatch: pytest.MonkeyPatch) -> pathlib.Path:
+    """Creates a temporary es-models directory and ensures via patching that it is
+    loaded by the resources library.
+    """
+    models = tmp_path / "es-models"
 
-@pytest.fixture
-def empty_descriptions_mappings(tmp_path):
-    foo_mapping, foo_dir_name = get_generic_mapping()
-    descriptions = {}
-    mapps = {
-        'foo': {
-            'mapping': foo_mapping,
-            'dir': foo_dir_name,
-            'descriptions': descriptions,
-        }
-    }
-    root = tmp_path
-    return mock_mappings(root, mapps)
+    models.mkdir()
```

Comment:
never removed?

Could `with tempfile... as t:` help?

https://docs.python.org/3.8/library/tempfile.html?highlight=tempfile#module-tempfile

Reply:
I was just using the pytest fixture tmp_path that had already been in use w/ the assumption that it is cleaned up like tempfile. But you are correct that pytest does not clean these files up. I'll update things to use tmpfile.

Reply:
Thanks!